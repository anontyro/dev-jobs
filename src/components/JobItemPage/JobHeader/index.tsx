import React from "react";
import styled from "@emotion/styled";
import { JobCardBody } from "../../homePage/JobList";
import JobQueryData from "../../../interfaces/jobQueryData";

interface JobPageHeaderContainer {
  backgroundColor: string;
}

const JobPageHeaderContainer = styled.div`
  display: flex;
  width: 100%;
  height: 8rem;
  padding: 0;
  margin-top: -2.5rem;
  margin-bottom: 2.5rem;
  .company-name {
    background-color: ${(props: JobPageHeaderContainer) =>
      props.backgroundColor};
    color: white;
    flex-shrink: 0;
    width: 8rem;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 10px 0 0 10px;
    font-size: 1.5rem;
    text-transform: lowercase;
  }
  .company-detail {
    display: flex;
    flex-grow: 1;
    padding: 2rem;
    justify-content: space-between;
  }
  .company-detail-buttons {
  }
  @media (max-width: 768px) {
  }
  @media (max-width: 425px) {
    height: 10rem;
    flex-direction: column;
    margin-top: -0.5rem;
    position: relative;
    margin-bottom: 1rem;
    .company-name {
      position: absolute;
      top: -2.5rem;
      width: 4rem;
      left: 42%;
      height: 4rem;
      border-radius: 10px;
      font-size: 1rem;
      font-weight: 400;
    }
    .company-detail {
      padding: 2rem;
      flex-direction: column;
      text-align: center;
    }
  }
`;

const JobItemCardBody = styled(JobCardBody)`
  @media (max-width: 768px) {
  }
  @media (max-width: 425px) {
    height: auto;
  }
`;

interface JobPageProps {
  job: JobQueryData;
}

const JobHeader: React.FC<JobPageProps> = ({ job }) => (
  <JobPageHeaderContainer
    className="card"
    backgroundColor={job.company.backgroundColor}
  >
    <div className="company-name boldest">
      <span>{job.company.name}</span>
    </div>
    <div className="company-detail">
      <JobItemCardBody>
        <h3>{job.company.name}</h3>
        <p className="card-text-muted">{job.company.url}</p>
      </JobItemCardBody>
      <div className="company-detail-buttons flex-center-items">
        <button className="button light">Company Site</button>
      </div>
    </div>
  </JobPageHeaderContainer>
);

export default JobHeader;
