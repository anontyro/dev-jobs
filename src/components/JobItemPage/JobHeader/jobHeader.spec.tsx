import React from "react";
import renderer from "react-test-renderer";
import JobHeader from ".";
import JobQueryData from "../../../interfaces/jobQueryData";

describe("JobHeader", () => {
  it("renders correctly", () => {
    const job = {
      dateTimeAdded: "1630116426229",
      type: "Full Time",
      title: "Software Engineer",
      locations: ["London"],
      company: {
        name: "Big Blue",
        url: "https://somesite.somewhere",
        backgroundColor: "#ffffff",
        description: "We are a small team of software engineers",
      },
      details: {},
    } as unknown as JobQueryData;
    const tree = renderer.create(<JobHeader job={job} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
