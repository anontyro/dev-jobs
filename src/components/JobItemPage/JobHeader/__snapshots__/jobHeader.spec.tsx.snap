// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobHeader renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-aew58a{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;width:100%;height:8rem;padding:0;margin-top:-2.5rem;margin-bottom:2.5rem;}.css-aew58a .company-name{background-color:#ffffff;color:white;-webkit-flex-shrink:0;-ms-flex-negative:0;flex-shrink:0;width:8rem;display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;-webkit-justify-content:center;justify-content:center;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;border-radius:10px 0 0 10px;font-size:1.5rem;text-transform:lowercase;}.css-aew58a .company-detail{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex:1;-webkit-flex-grow:1;-ms-flex-positive:1;flex-grow:1;padding:2rem;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;}@media (max-width: 425px){.css-aew58a{height:10rem;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;margin-top:-0.5rem;position:relative;margin-bottom:1rem;}.css-aew58a .company-name{position:absolute;top:-2.5rem;width:4rem;left:42%;height:4rem;border-radius:10px;font-size:1rem;font-weight:400;}.css-aew58a .company-detail{padding:2rem;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;text-align:center;}}",
      }
    }
    data-emotion="css aew58a"
  />,
  <div
    className="card css-aew58a"
  >
    <div
      className="company-name boldest"
    >
      <span>
        Big Blue
      </span>
    </div>
    <div
      className="company-detail"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1rny88i{height:5rem;}.css-1rny88i h3{margin:0;}.css-1rny88i p{margin:0;margin-top:5px;}@media (max-width: 425px){.css-1rny88i{height:auto;}}",
          }
        }
        data-emotion="css 1rny88i"
      />
      <div
        className="css-1rny88i"
      >
        <h3>
          Big Blue
        </h3>
        <p
          className="card-text-muted"
        >
          https://somesite.somewhere
        </p>
      </div>
      <div
        className="company-detail-buttons flex-center-items"
      >
        <button
          className="button light"
        >
          Company Site
        </button>
      </div>
    </div>
  </div>,
]
`;
